File: Arrays/09-maximum_consecutive_ones/maximum_consecutive_ones.py
from typing import List

# Given an array, find the maximum number of consecutive ones in the array

'''
Brute Force Approach
---------------------
LOGIC - Take a final_count and a temp_count, loop over the array, and store the count of the ones in a temp_count and if a number comes which
is not one then, check if the temp count is greater than the final count, if yes update the final count else pass on.

TIME COMPLEXITY - O(N)
SPACE COMPLEXITY - O(1)
'''

def maximum_consecutive_ones_brute( array : List[int] ):
    final_count = 0
    temp_count = 0
    for num in array:
        if num == 1: temp_count+=1
        else:
            final_count = max(final_count, temp_count)
            temp_count = 0
    return max(final_count, temp_count)

File: Arrays/09-maximum_consecutive_ones/maximum_consecutive_ones.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <set>
using namespace std;

// Given an array, find the maximum number of consecutive ones in the array

/*
Brute Force Approach
---------------------
LOGIC - 

TIME COMPLEXITY - O(N^2)
SPACE COMPLEXITY - O(1)
*/

int maximum_consecutive_ones_brute( vector<int> array ){
    int n = array.size();
    if (n == 0) return -1;

    int final_count = 0;
    int temp_count = 0;
    for (auto num: array){
        if (num == 1) temp_count++;
        else {
            if (temp_count > final_count) final_count = temp_count;
            temp_count = 0;
        }
    }
    return final_count;
}

